diff -ur openssl-1.1.0f/Configurations/10-main.conf ../openssl-1.1.0f/Configurations/10-main.conf
--- openssl-1.1.0f/Configurations/10-main.conf	2017-05-25 05:46:17.000000000 -0700
+++ ../openssl-1.1.0f/Configurations/10-main.conf	2017-09-19 21:57:44.308358429 -0700
@@ -910,7 +910,7 @@
         # systems are perfectly capable of executing binaries targeting
         # Froyo. Keep in mind that in the nutshell Android builds are
         # about JNI, i.e. shared libraries, not applications.
-        cflags           => add(picker(default => "-mandroid -fPIC --sysroot=\$(CROSS_SYSROOT) -Wa,--noexecstack")),
+        cflags           => add(picker(default => "-fPIC --sysroot=\$(CROSS_SYSROOT) -Wa,--noexecstack")),
         bin_cflags       => "-pie",
     },
     "android-x86" => {
@@ -950,7 +950,7 @@
 
     "android64" => {
         inherit_from     => [ "linux-generic64" ],
-        cflags           => add(picker(default => "-mandroid -fPIC --sysroot=\$(CROSS_SYSROOT) -Wa,--noexecstack")),
+        cflags           => add(picker(default => "-fPIC --sysroot=\$(CROSS_SYSROOT) -Wa,--noexecstack")),
         bin_cflags       => "-pie",
     },
     "android64-aarch64" => {
